src/Config/ConfigUpdater.php

```diff
@@ -0,0 +1,276 @@
+   *
+   * @var string[]
+   */
+  protected $typesByPrefix = [];
+
+  /**
+   * Constructs a new ConfigUpdater object.
+   *
+   * @param \Drupal\Core\Entity\EntityTypeManagerInterface $entity_type_manager
+   *   The entity type manager.
+   * @param \Drupal\Core\Config\StorageInterface $active_config_storage
+   *   The active config storage.
+   * @param \Drupal\Core\Config\ConfigFactoryInterface $config_factory
+   *   The config factory.
+   */
+  public function __construct(EntityTypeManagerInterface $entity_type_manager, StorageInterface $active_config_storage, ConfigFactoryInterface $config_factory) {
+    $this->entityTypeManager = $entity_type_manager;
+    $this->activeConfigStorage = $active_config_storage;
+    $this->extensionConfigStorage = new ExtensionInstallStorage($active_config_storage, InstallStorage::CONFIG_INSTALL_DIRECTORY);
+    $this->extensionOptionalConfigStorage = new ExtensionInstallStorage($active_config_storage, InstallStorage::CONFIG_OPTIONAL_DIRECTORY);
+    $this->configFactory = $config_factory;
+
+    foreach ($this->entityTypeManager->getDefinitions() as $entity_type => $definition) {
+      if ($definition->isSubclassOf('Drupal\Core\Config\Entity\ConfigEntityInterface')) {
+        /** @var \Drupal\Core\Config\Entity\ConfigEntityTypeInterface $definition */
+        $prefix = $definition->getConfigPrefix();
+        $this->typesByPrefix[$prefix] = $entity_type;
+      }
+    }
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function import(array $config_names) {
+    $succeeded = [];
+    $failed = [];
+
+    foreach ($config_names as $config_name) {
+      // Read the config from the file.
+      $value = $this->loadFromExtension($config_name);
+      if (!$value) {
+        $failed[$config_name] = $this->t('@config did not exist in extension storage', ['@config' => $config_name]);
+        continue;
+      }
+
+      if ($this->loadFromActive($config_name)) {
+        $failed[$config_name] = $this->t('@config already exists, use revert to update', ['@config' => $config_name]);
+        continue;
+      }
+
+      $type = $this->getConfigType($config_name);
+
+      // Save it as a new config entity or simple config.
+      if ($type == 'system.simple') {
+        $this->configFactory->getEditable($config_name)->setData($value)->save();
+      }
+      else {
+        /** @var \Drupal\Core\Config\Entity\ConfigEntityStorageInterface $entity_storage */
+        $entity_storage = $this->entityTypeManager->getStorage($type);
+        $entity = $entity_storage->createFromStorageRecord($value);
+        $entity->save();
+      }
+
+      $succeeded[$config_name] = $this->t('@config was successfully imported', ['@config' => $config_name]);
+    }
+
+    return new ConfigUpdateResult($succeeded, $failed);
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function revert(array $config_names, $skip_modified = TRUE) {
+    $succeeded = [];
+    $failed = [];
+
+    foreach ($config_names as $config_name) {
```

Comment:
Doesn't check that the config exists in active storage (which means that the loaded $entity could be NULL, and getEditable() might be creating a new object by accident).